### PR TITLE
[Parameter] Deprecate names starting or ending with a colon

### PR DESCRIPTION
**Pull request by fancyweb**
As requested by @Majkl578 in https://github.com/doctrine/doctrine2/pull/5996

I wasn't able to make a test as I don't know how to test a deprecation message without the @expectedDeprecation annotation of the Symfony PHPUnit Bridge ^^

### CODE REVIEW

**Comment by Ocramius** [comment]
Requires a test case

**Comment by fancyweb** [comment]
@Ocramius As I said in my comment I need help for this test.

**Comment by coudenysj** [comment]
@fancyweb Just construct the class with a faulty string, and watch what happens.

> By default, PHPUnit will install an error handler that converts the following errors to exceptions: https://phpunit.de/manual/current/en/appendixes.configuration.html

**Comment by fancyweb** [comment]
I tried that and it was just marked as a risky test since there was no assertion. No exception was thrown.

**Comment by Majkl578** [comment]
Please wait until #6869 is merged (after 2.6 is released), it will add the infrastructure for testing deprecations.

**Comment by lcobucci** [comment]
Just missing the test so I'll keep on 2.7.0 and we can easily test it with things added in #7901

**Comment by lcobucci** [comment]
@fancyweb thanks, I've updated things to add the test. I'm just struggling on explaining the value of this deprecation in `UPGRADING.md`. I'll move this 2.8.0 so we have enough time to discuss this :+1:

**Comment by SenseException** [comment]
Thanks for adding the tests.

> I'm just struggling on explaining the value of this deprecation in `UPGRADING.md`

Maybe describe it as a clear information about how a parameter has to look like and being less ambiguous?

**Comment by lcobucci** [comment]
@SenseException focusing on the ambiguity is a good idea. I guess what made me hesitate was also doing the check using regex. We could do the trim and then compare with the original value, sending the deprecation notice when they are different.

File: lib/Doctrine/ORM/Query/Parameter.php
Comment (greg0ire) [inline lib/Doctrine/ORM/Query/Parameter.php]: Let's add the package name and use some linebreaks:
```suggestion
            @trigger_error(
                'Starting or ending a parameter name with ":" is deprecated since Doctrine ORM 2.7 and will cause an error in 3.0',
                E_USER_DEPRECATED
            );
```

File: tests/Doctrine/Tests/ORM/Query/ParameterTest.php
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Query/ParameterTest.php]:
```suggestion
        $this->expectDeprecationMessage('Starting or ending a parameter name with ":" is deprecated since Doctrine ORM 2.7 and will cause an error in 3.0');
```

File: tests/Doctrine/Tests/ORM/Query/ParameterTest.php
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Query/ParameterTest.php]:
```suggestion
        $this->expectDeprecationMessage('Starting or ending a parameter name with ":" is deprecated since Doctrine ORM 2.7 and will cause an error in 3.0');
```

File: tests/Doctrine/Tests/ORM/Query/ParameterTest.php
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Query/ParameterTest.php]:
```suggestion
        $this->expectDeprecationMessage('Starting or ending a parameter name with ":" is deprecated since Doctrine ORM 2.7 and will cause an error in 3.0');
```

File: tests/Doctrine/Tests/ORM/Query/ParameterTest.php
Comment (lcobucci) [reply tests/Doctrine/Tests/ORM/Query/ParameterTest.php]: It should be 2.8 too :)

**Review by SenseException — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
Can we pick this up for 2.13?

**Pull request closed without merging**
